src/utils/tabs.rs

```diff
@@ -0,0 +1,74 @@
+use crate::config::SPACES;
+
+use unicode_segmentation::UnicodeSegmentation;
+
+pub fn clamp_arg(tab_size: usize) -> usize {
+    std::cmp::min(tab_size, SPACES.len())
+}
+
+/// Expand tabs as spaces.
+/// tab_width = 0 is documented to mean do not replace tabs.
+pub fn expand(line: &str, tab_width: usize) -> String {
+    let line_bytes = line.as_bytes();
```

Comment:
Why `as_bytes()` and not `chars()`.
Using chars from the beginning may allow the code not to check again `std::str::from_utf8(&expanded)` at the end of the expansion.
If it was valid before expansion, it will still be after it.

Reply:
I was using bytes directly because I only needed ascii, and not using `chars()` _might_ be faster _if_ I also do the unsafe bytes -> str conversion. But I don't want to use unsafe unless a benchmark points to this as a bottleneck which is very unlikely. So yes, I could have used `chars()` right from the beginning.